### PR TITLE
chapter 3: Mostly Static Pages

### PR DESCRIPTION
**Pull request by hazwanizafirah**
ready

### CODE REVIEW

**Comment by hazwanizafirah** [comment]
ready

File: app/assets/stylesheets/feedbacks.scss [hunk 1]
```diff
@@ -0,0 +1,3 @@
+// Place all the styles related to the Feedbacks controller here.
+// They will automatically be included in application.css.
```
Comment (anlvq-1058) [inline app/assets/stylesheets/feedbacks.scss:2]: You can delete this file if you don't use it

File: app/assets/stylesheets/static.scss [hunk 2]
```diff
@@ -0,0 +1,3 @@
+// Place all the styles related to the Static controller here.
```
Comment (anlvq-1058) [inline app/assets/stylesheets/static.scss:1]: You can delete this file if you don't use it

File: app/assets/stylesheets/static_pages.scss [hunk 3]
```diff
@@ -0,0 +1,3 @@
+// Place all the styles related to the StaticPages controller here.
```
Comment (anlvq-1058) [inline app/assets/stylesheets/static_pages.scss:1]: You can delete this file if you don't use it

File: app/controllers/feedbacks_controller.rb [hunk 4]
```diff
@@ -0,0 +1,66 @@
+class FeedbacksController < ApplicationController
```
Comment (anlvq-1058) [inline app/controllers/feedbacks_controller.rb:1]: Why you have `FeedbacksController` ?

File: app/controllers/static_controller.rb [hunk 5]
```diff
@@ -0,0 +1,7 @@
+class StaticController < ApplicationController
```
Comment (anlvq-1058) [inline app/controllers/static_controller.rb:1]: Why you have `StaticController`?

File: app/controllers/static_pages_controller.rb [hunk 6]
```diff
@@ -0,0 +1,7 @@
+class StaticPagesController < ApplicationController
+  def home
+  end
+
+  def help
+  end
```
Comment (anlvq-1058) [inline app/controllers/static_pages_controller.rb:6]:
```suggestion
  def home; end

  def help; end
```

File: app/helpers/application_helper.rb [hunk 7]
```diff
@@ -1,2 +1,6 @@
 module ApplicationHelper
+  def full_title page_title = ""
+    base_title = "Ruby on Rails Tutorial Sample App"
```
Comment (anlvq-1058) [inline app/helpers/application_helper.rb:3]: please use i18n to translate "Ruby on Rails Tutorial Sample App"

File: app/helpers/feedbacks_helper.rb [hunk 8]
```diff
@@ -0,0 +1,2 @@
+module FeedbacksHelper
+end
```
Comment (anlvq-1058) [inline app/helpers/feedbacks_helper.rb:2]: You can delete this file if you don't use it

File: app/helpers/static_helper.rb [hunk 9]
```diff
@@ -0,0 +1,2 @@
+module StaticHelper
+end
```
Comment (anlvq-1058) [inline app/helpers/static_helper.rb:2]: You can delete this file if you don't use it

File: app/helpers/static_pages_helper.rb [hunk 10]
```diff
@@ -0,0 +1,2 @@
+module StaticPagesHelper
+end
```
Comment (anlvq-1058) [inline app/helpers/static_pages_helper.rb:2]: You can delete this file if you don't use it

File: app/views/layouts/_header.html.erb [hunk 11]
```diff
@@ -0,0 +1,13 @@
+<header class="navbar navbar-fixed-top navbar-inverse">
+  <div class="container">
+    <%= link_to t(:h_title) , "#", id: "logo" %>
+    <nav>
+      <ul class="nav navbar-nav navbar-right">
+      <li><%= link_to t(:btn_home), home_path %></li>
+      <li><%= link_to t(:help), help_path %></li>
+      <li><%= link_to t(:log_in), "#"%></li>
```
Comment (anlvq-1058) [inline app/views/layouts/_header.html.erb:8]:
```suggestion
      <li><%= link_to t(:log_in), "#" %></li>
```

**Review by anlvq-1058 — COMMENTED** [review]
cmt

**Comment by hazwanizafirah** [comment]
ready

**Comment by hazwanizafirah** [comment]
ready

File: app/assets/stylesheets/custom.scss [hunk 12]
```diff
@@ -0,0 +1,32 @@
+/* universal */
+body {
```
Comment (huongnt-2545) [inline app/assets/stylesheets/custom.scss:2]: Please fix indentation in this file, follow:

```suggestion
body {
  padding-top: 60px;
}

section {
  overflow: auto;
}

...
```

File: app/controllers/static_pages_controller.rb [hunk 13]
```diff
@@ -0,0 +1,7 @@
+class StaticPagesController < ApplicationController
+  def home
```
Comment (huongnt-2545) [inline app/controllers/static_pages_controller.rb:2]: Method content is empty:
```
# good
def home; end

# not good
def home
end
```

Please apply it

File: app/views/layouts/application.html.erb [hunk 14]
```diff
@@ -1,16 +1,23 @@
...
+
+    <body>
+    <%= render "layouts/header" %>
+      <div class="container">
+        <%= yield %>
+      </div>
+
+    </body>
+   </html>
+
```
Comment (huongnt-2545) [inline app/views/layouts/application.html.erb:22]: delete 2 empty lines 21, 22

File: app/views/static_pages/help.html.erb [hunk 15]
```diff
@@ -0,0 +1,12 @@
+<%= provide :title, "Help" %>
```
Comment (huongnt-2545) [inline app/views/static_pages/help.html.erb:1]: translate all text in this file using I18n (the same as `app/views/layouts/_header.html.erb` file)

File: config/application.rb [hunk 16]
```diff
@@ -11,12 +11,10 @@ class Application < Rails::Application
...
-    # in config/environments, which are processed later.
-    #
-    # config.time_zone = "Central Time (US & Canada)"
-    # config.eager_load_paths << Rails.root.join("extras")
+    config.i18n.available_locales = [:en, :ja]
+    config.i18n.default_locale = :en
+    config.i18n.fallbacks = true
   end
 end
+
```
Comment (huongnt-2545) [inline config/application.rb:19]: delete 2 empty line

File: app/models/feedback.rb [hunk 17]
```diff
@@ -0,0 +1,2 @@
+class Feedback < ApplicationRecord
```
Comment (huongnt-2545) [inline app/models/feedback.rb:1]: delete this file, it is not follow tutorial

File: config/routes.rb [hunk 18]
```diff
@@ -1,3 +1,8 @@
 Rails.application.routes.draw do
-  # For details on the DSL available within this file, see https://guides.rubyonrails.org/routing.html
+  scope "(:locale)", locale: /#{I18n.available_locales.join("|")}/ do
+    get 'home/index'
+    resources :posts
+     root 'static_pages#home', as: "home"
```
Comment (huongnt-2545) [inline config/routes.rb:5]: indent = 2 spaces
```suggestion
      root 'static_pages#home', as: "home"
```

File: db/migrate/20210821091830_create_feedbacks.rb [hunk 19]
```diff
@@ -0,0 +1,10 @@
+class CreateFeedbacks < ActiveRecord::Migration[6.1]
```
Comment (huongnt-2545) [inline db/migrate/20210821091830_create_feedbacks.rb:1]: delete this file, it is not follow tutorial

File: test/controllers/feedbacks_controller_test.rb [hunk 20]
```diff
@@ -0,0 +1,48 @@
+require "test_helper"
```
Comment (huongnt-2545) [inline test/controllers/feedbacks_controller_test.rb:1]: delete this file

**Comment by hazwanizafirah** [comment]
ready

File: app/assets/stylesheets/custom.scss [hunk 21]
```diff
@@ -0,0 +1,36 @@
...
+  text-align: center;
+}
+
+.center h1 {
+  margin-bottom: 10px;
+}
+
+/* header */
+#logo {
+float: left;
```
Comment (huongnt-2545) [inline app/assets/stylesheets/custom.scss:24]: fix indentation in this block like the same above

File: app/assets/stylesheets/custom.scss [hunk 22]
```diff
@@ -0,0 +1,36 @@
...
+margin-right: 10px;
+font-size: 1.7em;
+color: #fff;
+text-transform: uppercase;
+letter-spacing: -1px;
+padding-top: 9px;
+font-weight: bold;
+}
+#logo:hover {
+color: #fff;
```
Comment (huongnt-2545) [inline app/assets/stylesheets/custom.scss:34]: the same

File: app/assets/stylesheets/scaffolds.scss [hunk 23]
```diff
@@ -0,0 +1,65 @@
+body {
+  background-color: #fff;
+  color: #333;
+  margin: 33px; }
```
Comment (huongnt-2545) [inline app/assets/stylesheets/scaffolds.scss:4]: good
```suggestion
  margin: 33px;
}
```

File: config/routes.rb [hunk 24]
```diff
@@ -1,3 +1,8 @@
 Rails.application.routes.draw do
-  # For details on the DSL available within this file, see https://guides.rubyonrails.org/routing.html
+  scope "(:locale)", locale: /#{I18n.available_locales.join("|")}/ do
+    get 'home/index'
+    resources :posts
+      root 'static_pages#home', as: "home"
```
Comment (huongnt-2545) [inline config/routes.rb:5]:
```suggestion
    root 'static_pages#home', as: "home"
```

File: test/fixtures/feedbacks.yml [hunk 25]
```diff
@@ -0,0 +1,9 @@
+# Read about fixtures at https://api.rubyonrails.org/classes/ActiveRecord/FixtureSet.html
```
Comment (huongnt-2545) [inline test/fixtures/feedbacks.yml:1]: remove all file related to feedback, these are not part of tutorial

**Comment by hazwanizafirah** [comment]
ready

**Review by huongnt-2545 — APPROVED** [review]
merged
You can merge this pull request

**Pull request merged**